qiskit/algorithms/time_evolvers/classical_methods/evolve.py

```diff
@@ -0,0 +1,217 @@
+    Args:
+        ops_ev_mean: Array containing the expectation value of each observable at each timestep.
+        evolution_problem: Time Evolution Problem to create the output of.
+
+    Returns:
+        An output with the observables mean value at the appropriate times depending on whether
+        the auxiliary operators in the time evolution problem are a `list` or a `dict`.
+
+    """
+
+    aux_ops = evolution_problem.aux_operators
+
+    time_array = np.linspace(0, evolution_problem.time, ops_ev_mean.shape[-1])
+    zero_array = np.zeros(ops_ev_mean.shape[-1])  # std=0 since it is an exact method
+
+    operators_number = 0 if aux_ops is None else len(aux_ops)
+
+    observable_evolution = [(ops_ev_mean[i], zero_array) for i in range(operators_number)]
+
+    if isinstance(aux_ops, dict):
+        observable_evolution = dict(zip(aux_ops.keys(), observable_evolution))
+
+    return observable_evolution, time_array
+
+
+def _create_obs_final(
+    ops_ev_mean: np.ndarray,
+    evolution_problem: TimeEvolutionProblem,
+) -> ListOrDict[tuple[complex, complex]]:
+    """Creates the right output format for the final value of the auxiliary operators.
+
+    Args:
+        ops_ev_mean: Array containing the expectation value of each observable at the final timestep.
+        evolution_problem: Evolution problem to create the output of.
+
+    Returns:
+        An output with the observables mean value at the appropriate times depending on whether
+        the auxiliary operators in the evolution problem are a `list` or a `dict`.
+
+    """
+
+    aux_ops = evolution_problem.aux_operators
+    aux_ops_evaluated = [(op_ev, 0) for op_ev in ops_ev_mean]
+    if isinstance(aux_ops, dict):
+        aux_ops_evaluated = dict(zip(aux_ops.keys(), aux_ops_evaluated))
+    return aux_ops_evaluated
+
+
+def _evaluate_aux_ops(
+    aux_ops: list[csr_matrix],
+    state: np.ndarray,
+) -> tuple[np.ndarray, np.ndarray]:
+    """Evaluates the aux operators if they are provided and stores their value.
+
+    Returns:
+        Tuple of the mean and standard deviation of the aux operators for a given state.
+    """
+    op_means = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+    return op_means
+
+
+def _operator_to_matrix(operator: BaseOperator | PauliSumOp):
+
+    if isinstance(operator, PauliSumOp):
+        op_matrix = operator.to_spmatrix()
+    else:
+        try:
+            op_matrix = operator.to_matrix(sparse=True)
+        except TypeError:
+            logger.debug(
+                "WARNING: operator of type `%s` does not support sparse matrices. "
+                "Trying dense computation",
+                type(operator),
+            )
+        try:
+            op_matrix = operator.to_matrix()
+        except AttributeError as ex:
+            raise AlgorithmError(f"Unsupported operator type `{type(operator)}`.") from ex
+    return op_matrix
+
+
+def _sparsify(
```

Comment:
Not blocking since it's a private function but the name `_sparsify` seems a bit misleading as actually `_operator_to_matrix` constructs the sparse matrix and this method here prepared all the inputs. How about renaming this to something like `_build_evolution_op` or just putting it inside the `evolve` method completely? 😄

Reply:
I went with `_build_scipy_operators` :smile: